### PR TITLE
Issue 855

### LINKED ISSUE #855: Make optional warmup_end_value in create_lr_scheduler_with_warmup
## 🚀 Feature

It would be interesting to setup a lr scheduler without specifying `warmup_end_value` which can be deduced from provided optimizer:
```python
import torch
from torch.optim.lr_scheduler import ExponentialLR

t1 = torch.tensor([0.0])
opt = torch.optim.SGD([t1], lr=1.0)

torch_lr_scheduler = ExponentialLR(optimizer=opt, gamma=0.98)
lr_values = [None] * 100
scheduler = create_lr_scheduler_with_warmup(torch_lr_scheduler,
                                            warmup_start_value=0.0,
                                            warmup_end_value=None,  # warmup_end_value -> 1.0
                                            warmup_duration=10,
                                            output_simulated_values=lr_values)

print(lr_values)
```

### PR DESCRIPTION
**Pull request by ykumards**
Fixes #855

Description: When warmup_end_value is None, it is set to optimizer initial LR

Check list:
* [x] New tests are added (if a new feature is added)
* [ ] New doc strings: description and/or example code are in RST format
* [x] Documentation is updated (if required)

### CODE REVIEW

**Comment by ykumards** [comment]
> @ykumards Please, don't forget to apply black on sources :) I can't commit in your repo to do it.
> 
> LGTM :)

Oh, we had a hook for that right?

**Comment by sdesrozis** [comment]
> > @ykumards Please, don't forget to apply black on sources :) I can't commit in your repo to do it.
> > LGTM :)
> 
> Oh, we had a hook for that right?

Oh maybe during merge because it missed something here. Btw, that's not really important and it was by pure chance that I saw that :)

**Comment by ykumards** [comment]
> > > @ykumards Please, don't forget to apply black on sources :) I can't commit in your repo to do it.
> > > LGTM :)
> > 
> > 
> > Oh, we had a hook for that right?
> 
> Oh maybe during merge because it missed something here. Btw, that's not really important and it was by pure chance that I saw that :)

@sdesrozis You were right! Black formatting was off, thanks pointing it out :)
